### PR TITLE
Fix attached accounts race condition

### PR DESCRIPTION
**Pull request by ejizba**
I saw issues where the connected mongo database didn't always persist for attached accounts. It's because there is a race condition related to loading the persisted accounts.

I suppose there was always a bit of a race condition here. For example, if the user expanded the 'Attached Accounts' node before we load the persisted accounts, they would not see their accounts.

### CODE REVIEW

**Comment by EricJizbaMSFT** [comment]
I would only be nervous if we'd already done a bunch of testing. We have to test anyways, so I don't see the harm in fixing this

**Comment by nturinski** [comment]
I tried my hardest to break it.

File: src/tree/AttachedAccountsTreeItem.ts [hunk 1]
```diff
@@ -32,12 +32,27 @@ export class AttachedAccountsTreeItem implements IAzureParentTreeItem {
...
 
+    private _loadPersistedAccountsTask: Promise<IAzureTreeItem[]>;
+
     constructor(private readonly _globalState: vscode.Memento) {
         this._keytar = fetchNodeModule('keytar');
-        this.loadPersistedServers();
+        this._loadPersistedAccountsTask = this.loadPersistedAccounts();
+    }
+
+    private async getAttachedAccounts(): Promise<IAzureTreeItem[]> {
```
Comment (PrashanthCorp) [inline src/tree/AttachedAccountsTreeItem.ts]: Nit: Can we move this function to below loadMoreChildren? :P 
If I understand correctly, that's the only call site, and that makes it easier to read. 
Feel free to ignore.

Comment (EricJizbaMSFT) [reply src/tree/AttachedAccountsTreeItem.ts]: It's also called in 'attachAccount' and 'detach'

**Pull request merged**